src/api-requests/get-all-search-content.ts

```diff
@@ -1,12 +1,46 @@
-import axios from 'axios';
+import {
+    buildSearchQuery,
+    SearchQueryParams,
+} from '../components/search/utils';
 import { SearchItem } from '../components/search/types';
 
 // Should find a way to cache this response. Only use it in a few places for search filters but still.
 export const getAllSearchContent = async (): Promise<SearchItem[]> => {
-    return axios
-        .get(`${process.env.REALM_SEARCH_URL}/search_devcenter?s=`)
-        .then(async response => {
-            const r_json: SearchItem[] = response.data;
+    const url = `${process.env.REALM_SEARCH_URL}/search_devcenter?s=`;
+    const options = {
+        method: 'GET',
+        headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+        },
+    };
+
+    return fetch(url, options)
```

Comment:
if we're moving away from axios, we might want to add a safeguard here to handle failures, and we can also utilize `async/await` since it's already declared on the function

```
try {
   const req = await fetch(url, options);
   // do if (req.status === 200 || req.ok) check if you want
   const { data } = await req.json();
   return data;
} catch (e) { ... }
```